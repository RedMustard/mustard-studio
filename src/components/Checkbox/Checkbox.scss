@import '../../style/variables.scss';

.checkbox {
    display: inline-block;
    height: 14px;
    position: relative;
    width: 5%;
}

.checkbox__input-like {
    border: 1px solid #fff;
    height: 13px;
    left: 0;
    position: absolute;
    top: 0;
    transition: .2s;
    width: 13px;
}

.checkbox__input-like::after {
    background-color: $green;
    border-width: 0 2px 2px 0;
    content: '';
    display: none;
    height: 7px;
    left: 2px;
    position: absolute;
    top: 2px;
    transition: .2s;
    width: 7px;
}

.checkbox__input {
    cursor: pointer;
    margin: 0;
    opacity: 0;
    position: absolute;
    top: 0;
    z-index: 1;
}

.checkbox__input:checked ~ .checkbox__input-like {
    background-color: #000;
    transition: .2s;
}

.checkbox__input:focus ~ .checkbox__input-like {
    border: 1px solid #fff;
}

.checkbox__input:checked:focus ~ .checkbox__input-like {
    background-color: #000;
}

.checkbox__input:checked ~ .checkbox__input-like::after {
    display: block;
}
