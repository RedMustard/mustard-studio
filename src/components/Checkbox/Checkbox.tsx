import { h } from 'preact';


interface CheckboxProps {
    onInput: (isChecked: boolean) => void;
    isChecked: boolean;
}

export const Checkbox = ({ onInput, isChecked }: CheckboxProps) => (
    <div className="checkbox">
        <input
            className="checkbox__input"
            id="sid-opt-in"
            name="sid-opt-in"
            type="checkbox"
            checked={isChecked}
            onInput={(e) => {
                onInput(e.currentTarget.checked);
            }}
        />
        <span className="checkbox__input-like" />
    </div>
);
