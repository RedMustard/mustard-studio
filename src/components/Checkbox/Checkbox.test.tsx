import { h } from 'preact';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Checkbox } from './Checkbox';


const baseProps = {
    isChecked: false,
    onInput: jest.fn(),
};

describe('<Checkbox />', () => {
    it('renders with basic props', () => {
        const wrapper = shallow(
            <Checkbox
                {...baseProps}
            />,
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('renders checked', () => {
        const wrapper = shallow(
            <Checkbox
                {...baseProps}
                isChecked
            />,
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('calls onInput when changed', () => {
        const wrapper = mount(
            <Checkbox
                {...baseProps}
            />,
        );
        wrapper.find('.checkbox__input').simulate('input');
        expect(baseProps.onInput).toBeCalled();
    });
});
