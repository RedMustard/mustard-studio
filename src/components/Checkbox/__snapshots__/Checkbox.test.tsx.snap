// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> renders checked 1`] = `
<div
  className="checkbox"
>
  <input
    checked={true}
    className="checkbox__input"
    id="sid-opt-in"
    name="sid-opt-in"
    onInput={[Function]}
    type="checkbox"
  />
  <span
    className="checkbox__input-like"
  />
</div>
`;

exports[`<Checkbox /> renders with basic props 1`] = `
<div
  className="checkbox"
>
  <input
    checked={false}
    className="checkbox__input"
    id="sid-opt-in"
    name="sid-opt-in"
    onInput={[Function]}
    type="checkbox"
  />
  <span
    className="checkbox__input-like"
  />
</div>
`;
